module/VuFindConsole/tests/unit-tests/src/VuFindTest/Command/Util/IndexReservesCommandTest.php

```diff
@@ -330,9 +350,260 @@ public function testSuccessWithILS()
+                . '<add>'
+                . '<doc>'
+                . '<field name="id">course1|inst1|</field>'
+                . '<field name="bib_id">1</field>'
+                . '<field name="instructor_id">inst1</field>'
+                . '<field name="instructor">inst1</field>'
+                . '<field name="course_id">course1</field>'
+                . '<field name="course">course1</field>'
+                . '<field name="department_id"></field>'
+                . '<field name="department">no_department_listed</field>'
+                . '</doc>'
+                . '</add>';
+            $that->assertEquals($expectedXml, trim($update->getContent()));
+            return true;
+        };
+        $writer->expects($this->once())->method('save')
+            ->with(
+                $this->equalTo('SolrReserves'),
+                $this->callback($updateValidator)
+            );
+        $command = $this->getCommand($writer, $ils);
+        $commandTester = new CommandTester($command);
+        $commandTester->execute([]);
+        $this->assertEquals(0, $commandTester->getStatusCode());
+        $this->assertStringContainsString(
+            'Successfully loaded 1 rows.',
             $commandTester->getDisplay()
         );
+        $this->assertStringNotContainsString(
+            'WARNING',
+            $commandTester->getDisplay()
+        );
+    }
+
+    /**
+     * Test successful ILS loading with some missing required fields
+     *
+     * @return void
+     */
+    public function testSuccessWithILSWithMissingRequiredData()
+    {
+        $ils = $this->getMockIlsConnection();
+        $instructors = ['inst1' => 'inst1'];
+        $courses = ['course1' => 'course1'];
+        $departments = ['dept1' => 'dept1'];
+        $reserves = [
+            [
+                'BIB_ID' => 1,
+                'COURSE_ID' => 'course1',
+                'INSTRUCTOR_ID' => 'inst1',
+            ],
+        ];
+        $this->expectConsecutiveCalls(
+            $ils,
+            '__call',
+            [
+                ['getInstructors'],
+                ['getCourses'],
+                ['getDepartments'],
+                ['findReserves'],
+            ],
+            [
+                $instructors,
+                $courses,
+                $departments,
+                $reserves,
+            ]
+        );
+
+        $writer = $this->getMockSolrWriter();
+        $that = $this;
+        $command = $this->getCommand($writer, $ils);
+        $commandTester = new CommandTester($command);
+        $this->expectException(
+            \Exception::class
+        );
+        $this->expectExceptionMessage(
+            'fields not present in reserve records. Please update ILS driver.'
+        );
+        $commandTester->execute([]);
+    }
+
+    /**
+     * Test unsuccessful ILS call that raises exception
+     *
+     * @return void
+     */
+    public function testILSExcetpion()
```

Comment:
Typo in method name; should be:

```suggestion
    public function testILSException()
```